### PR TITLE
ccli_test dial options for unencrypted gRPC and sending metadata

### PR DESCRIPTION
**Pull request by liulk**

### CODE REVIEW

File: cmd/ccli/ccli_test.go [hunk 1]
```diff
@@ -15,37 +15,80 @@
...
+	"google.golang.org/grpc/credentials"
+
+	spb "github.com/openconfig/gribi/v1/proto/service"
 )
 
 var (
-	addr = flag.String("addr", "", "address of the gRIBI server in the format hostname:port")
+	addr       = flag.String("addr", "", "address of the gRIBI server in the format hostname:port")
+	insecure   = flag.Bool("insecure", false, "dial insecure gRPC (no TLS)")
+	skipverify = flag.Bool("skipverify", true, "allow self-signed TLS certificate")
```
Comment (robshakir) [inline cmd/ccli/ccli_test.go:37]: (nit) Add something here that says that this is not required when "insecure" is also set?

File: cmd/ccli/ccli_test.go [hunk 2]
```diff
@@ -15,37 +15,80 @@
...
+type flagCred struct{}
+
+func (flagCred) GetRequestMetadata(ctx context.Context, uri ...string) (map[string]string, error) {
+	return map[string]string{
+		"username": *username,
+		"password": *password,
+	}, nil
+}
+
+func (flagCred) RequireTransportSecurity() bool {
```
Comment (robshakir) [inline cmd/ccli/ccli_test.go:51]: This looks to be unused. I presume it's required to implement the argument for `grpc.WithPerRPCCredentials`? I'd suggest to add a comment to `flagCred` that just states what interface it is implementing.

**Review by robshakir — APPROVED** [review]
LGTM, a couple of minor suggestions.

File: cmd/ccli/ccli_test.go (hunk 1 above)
Comment (liulk) [reply cmd/ccli/ccli_test.go]: Good idea! Done.

File: cmd/ccli/ccli_test.go (hunk 2 above)
Comment (liulk) [reply cmd/ccli/ccli_test.go:55]: That's right. Added comment to clarify.

**Pull request merged**